src/ui/components/address-action/AddressActionDetails/AddressActionDetails.tsx

```diff
@@ -28,18 +29,21 @@ export function AddressActionDetails({
   actionTransfers?: ActionTransfers;
   singleAsset?: NonNullable<AddressAction['content']>['single_asset'];
   allowanceQuantityBase: string | null;
+  showApplicationLine: boolean;
   singleAssetElementEnd: React.ReactNode;
 }) {
+  const showNetworkIcon = !(showApplicationLine && addressAction?.label);
   return (
     <>
       {recipientAddress && addressAction?.type.value === 'send' ? (
         <RecipientLine
           recipientAddress={recipientAddress}
           chain={chain}
+          showNetworkIcon={showNetworkIcon}
           networks={networks}
         />
       ) : null}
-      {addressAction?.label && addressAction?.label.type !== 'to' ? (
+      {showApplicationLine && addressAction?.label ? (
         <ApplicationLine
```

Comment:
I think this is all kinda backwards

It should be

```
const applicationLineVisible = Boolean(showApplicationLine && addressAction?.label);
...

              showNetworkIcon={!applicationLineVisible}
...

        {showApplicationLine && addressAction?.label ? ... : ...}
```

Reply:
agree, your suggestion looks more readable, thank you!